model/project_parser_db.go

```diff
@@ -65,6 +66,36 @@ func ParserProjectFindOne(query db.Q) (*ParserProject, error) {
 	return project, err
 }
 
+// ParserProjectByVersion finds a parser project with a given version.
+// If version is empty the last known good config will be returned
+func ParserProjectByVersion(projectId string, version string) (*ParserProject, error) {
+	lookupVersion := false
+	if version == "" {
+		lastGoodVersion, err := FindVersionByLastKnownGoodConfig(projectId, -1)
+		if err != nil || lastGoodVersion == nil {
+			grip.Error(message.WrapError(err, message.Fields{
+				"message":    fmt.Sprintf("Unable to retrieve last good version for project '%s'", projectId),
```

Comment:
Sorry for not having this thought the first time around: wrapping the message here is a bit unnecessary since projectId is listed in the fields; I would just put the "unable to retrieve last good version for project" bit (but it would be good to wrap the actual error returned at 81)

Reply:
same below